erpnext/manufacturing/doctype/work_order/work_order.py

```diff
@@ -795,3 +796,43 @@ def update_item_quantity(source, target, source_parent):
 	doc.set_item_locations()
 
 	return doc
+
+@frappe.whitelist()
+def start_job_cards(work_order):
+	job_cards = start_and_stop_job_cards(work_order, job_started=True)
+	return job_cards
+
+@frappe.whitelist()
+def stop_job_cards(work_order):
+	job_cards = start_and_stop_job_cards(work_order, job_started=False)
+	return job_cards
+
+def start_and_stop_job_cards(work_order, job_started=None):
+	job_cards = []
+	open_job_cards = frappe.db.get_all('Job Card',
+		filters={
+			'work_order': work_order,
+			'status': ('in', ['Open', 'Work In Progress', 'Material Transferred'])
+		})
+	for job_card in open_job_cards:
+		job = frappe.get_doc('Job Card', job_card)
+		job.job_started = job_started
+		if job_started == True:
+			existing_from_time_count = 0
+			for row in job.time_logs:
+				if not row.to_time:
+					existing_from_time_count = existing_from_time_count + 1
+					continue
+			if not existing_from_time_count:
+				row = job.append('time_logs', {
+					"from_time": now_datetime(),
+					"completed_qty": 0
+				})
```

Comment:
We don't need variables actually, this can be solved with `break` to go back to the outer loop

```suggestion
			for row in job.time_logs:
				if not row.to_time:
					break
				job.append('time_logs', {
					"from_time": now_datetime(),
					"completed_qty": 0
				})
```